### ISSUE TITLE
Code 0091A and 0011A are now mapped to both Drug and Procedure domains due to which we are noticing a huge increase in Drug Exposure  and Procedure records.

### ISSUE THREAD

**Issue by revettk** [OHDSI/Vocabulary-v5.0#719]
We are seeing very large increases in Moderna and Pfizer vaccination doses in the November ETL, relative to the October ETL process.
The result are increases of > 10,000% in vaccine doses that are clearly not accurate.

The 'drug_source_value'_ and  'procedure_source_values' - 0091A and 0011A are both being mapped to the same domains - Drug and Procedure.  These are CPT4 code mappings - RxNORM is mapping a drug_code to a procedure code - which is which we are seeing large increases in procedure values relative to previous ETLs.

I attach a screenshot to provide additional information:
.
![Nov_VOCAB_CPT4_ISSUE](https://user-images.githubusercontent.com/11462104/202748247-28faa4b9-7b3f-4c98-a102-b7d3faa00ce5.PNG)

We are asking if the RXNORM codes for 0091 and 0011A be mapped to their respective domains - 0091A is a CPT4 code for a Procedure to deliver the Covid19 vaccine. .  It is also being deployed as a drug (Covid19 vaccine). 
So, in short, 0091A and 0011A are being mapped to BOTH the Procedure and Drug domains.

The concept code 0091A should only be mapped to Procedure and 0011A should only be mapped to Drug.

@mik-ohdsi @dimshitc @cgreich

**Comment by mik-ohdsi** [comment]
Hi @revettk ,
the mapping convention you are seeing is actually representing the fact that from a procedure code that describes administration of a *specific* vaccine, we map to the procedure itself AND the vaccine / drug. This is to make sure that if only this procedure is coded, we still cover the drug exposure to that vaccine. If in the patient data, the vaccine itself is coded as well separately, a second drug exposure could theoretically be created. This kind of duplication is normally handled in the ETL process or can even be filtered on the analysis / data science step. I will call in my personal authority in all things ETL, @melaniephilofsky - what is your take and advice on this?

**Comment by don-torok** [comment]
Is this a change for CPT/HCPCS codes that map to a drug, that is,  mapping the code to both a procedure and a drug.  I believe that previously there would only be a single mapping to the drug.

**Comment by MelaniePhilofsky** [comment]
Hello @revettk  @mik-ohdsi @don-torok 

I took a look at these source codes. Per the AMA's guidelines for [0011A](https://www.ama-assn.org/system/files/cpt-assistant-guide-coronavirus-may-2022.pdf) and [0091A](https://www.ama-assn.org/system/files/cpt-assistant-guide-coronavirus-july-2022.pdf), these are both codes to be used for the "Procedure" of vaccine administration. These two codes need to be accompanied by the code '91309' which represents the actual "Drug" being given. Which, unfortunately in the case of 0011A loses the brand name of the drug. I think both 0011A and 0091A need to be mapped to a standard concept in the Procedure domain. 91309 maps correctly to a standard concept_id in the Drug domain.

As a FYI regarding EHR data folks should know, especially regarding Covid vaccines: These data are duplicated all over an EHR. We have records for "ordered" vaccines and then we have records for the "administered" vaccines. And almost everyone who gets a covid vaccine administered also has a covid vaccine ordered. I call orders and administrations "encounter records". Then there are the "billing records". These are the records which are CPT4 codes at the source. These are created to bill for the vaccine. So, a person has a record for an ordered vaccine, an administered vaccine and a billed vaccine. I have dug around enough to find a person can have more covid drug records than possible at this point in the pandemic. So, you shouldn't just look at # of Covid shots because this isn't accurate. Unfortunately, de-duplicating is a task for the analyst. And de-duplicating will depend on the source data.

**Comment by mik-ohdsi** [comment]
Thank you so much, @MelaniePhilofsky , for the insights!
Hey @revettk - my suggestion would be that we will re-evaluate that convention and (just for CPT4) would consider removing the drug mapping (but this needs a bit of discussion, as it affects all users of the CPT4 vocabulary).
Meanwhile, can you discuss de-duplication with your team?
Thanks - mik

**Comment by krevett-iqvia** [comment]
@MelaniePhilofsky @mik-ohdsi - so as it stands, '0011A and 0091A are essentially duplicates of one another.  Will you be considering mapping them to a standard concept in the Procedure domain?

**Comment by Alexdavv** [comment]
> @MelaniePhilofsky @mik-ohdsi - so as it stands, '0011A and 0091A are essentially duplicates of one another. Will you be considering mapping them to a standard concept in the Procedure domain?

The actual vaccine information goes to the Drug Domain.
A piece of whether it's a first/second/etc or a booster dose ends up in the Procedures (this is an ugly workaround but also the best we can do - the alternative is to drop this information).
The fact of administration (essentially an injection) means nothing in OMOP so goes nowhere.
"Intramuscular" could be mapped to the route_concept_id field but we don't do that because it would be another struggle for ETL.

**Comment by krevett-iqvia** [comment]
Hi @MelaniePhilofsky - thanks for this update.  we are attempting to de-dup - will see if that takes care of our issue in our current February ETL.  if so, I will update and we can potentially close this issue.  Thank you

**Comment by krevett-iqvia** [comment]
@MelaniePhilofsky 
@m-khitrun 
@mik-ohdsi 
I believe this issue can be closed?  Thank you for the effort on this ticket!
